ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PreJoinProjectNode.java

```diff
@@ -109,6 +123,12 @@ public void setKeyFormat(final KeyFormat format) {
     return super.validateColumns(builder.build());
   }
 
+  // Only safe to call this if joiningSource is empty.
+  private KeyFormat getSourceKeyFormat() {
```

Comment:
Would it be simpler to embed this code within `getPreferredKeyFormat`? It's not called from multiple places so what is the advantage of putting it into a method (and add a comment when it's safe to call?)

Reply:
Sure, I don't feel strongly.